apps/gitness/src/pages-v2/repo/repo-summary.tsx

```diff
@@ -163,7 +171,7 @@ export default function RepoSummaryPage() {
     }
 
     pathDetails({
-      queryParams: { git_ref: normalizeGitRef(selectedBranch.name) },
+      queryParams: { git_ref: normalizeGitRef(selectedBranch?.name || '') },
```

Comment:
at line 167 we are making the loading as true and we are checking for the condition `repoEntryPathToFileTypeMap.size`. We are setting the loader to false at line 196. If the condition not satisfies, we are returning from the function. Still loader is true and it will keep showing skeleton UI. 

It would be good if we place the entire effect within the condition.

```tsx
if(repoEntryPathToFileTypeMap.size){
  setLoading(true)

  ...

  .finally(() => {
      setLoading(false)
   })


}
```